### ISSUE TITLE
How to disable for a function?

### ISSUE THREAD

**Issue by josephzidell** [squizlabs/PHP_CodeSniffer#1794]
I need to disable cyclomatic complexity check for certain methods. In v <= 3.1.1, I did this:

```php
/**
 * Docblock here
 */
// @codingStandardsIgnoreStart
public function myMethod()
{
	// code here
}
// @codingStandardsIgnoreEnd
```

This wasn't good since it ignored everything in the method.

In v 3.2.0, I tried this:
```php
/**
 * Docblock here
 */
// phpcs:disable Generic.Metrics.CyclomaticComplexity
public function myMethod()
{
	// code here
}
// phpcs:enable
```

This worked for that check, but failed others:
- `Missing function doc comment`
- `Expected 1 blank line before function; 0 found`
- `Expected 1 blank line after function; 0 found`

Things I tried:
1. Putting it right above docblock
2. Putting it inside of the method

I also tried this:
```php
} // end of prev function

// phpcs:disable Generic.Metrics.CyclomaticComplexity

/**
 * Docblock here
 */
public function myMethod()
{
	// code here
}

// phpcs:enable

/* // next function
```
But I worry about git merges, whether having whitespace can allow git to insert another function between the `disable` and `enable`.

The generic question is: how to disable a check for a method?

**Comment by gsherwood** [comment]
In this specific case, the error is reported on the function declaration line, so you only need to ignore that line. I'd do something like this:
```php
<?php
/**
 * Description.
 *
 * phpcs:disable Generic.Metrics.CyclomaticComplexity
 *
 * @param ...
 * @return ...
 */
function foo()
{
    // phpcs:enable Generic.Metrics.CyclomaticComplexity
    ...
}
```
You can't tell PHPCS to ignore error for an entire function/class/etc block. Those comments are based on line numbers only.

**Comment by josephzidell** [comment]
This works. Thanks
